CMakeLists.txt

```diff
@@ -63,123 +74,41 @@ set(NOVELRT_VORBIS_VERSION "1.3.7" CACHE STRING "Vorbis version")
 set(NOVELRT_VULKAN_VERSION "1.3.231" CACHE STRING "Vulkan version")
 
 #
-# Allow for generating documentation
+# Configure Platforms
 #
-if(NOVELRT_BUILD_DOCUMENTATION)
-  find_package(Doxygen ${NOVELRT_DOXYGEN_VERSION}
-    COMPONENTS dot
-  )
-endif()
+include(NovelRTPlatformConfig)
 
 #
-# Alias "NovelRT::Utilities::Misc:Span" to "std::span" instead of "gsl::span".
-# If enabled, you need to make sure you current configuration supports "std::span".
+# Set default build type when not defined and not multi-config
 #
-if(NOVELRT_USE_STD_SPAN)
-  add_compile_definitions(NOVELRT_USE_STD_SPAN=true)
+if(NOT GENERATOR_IS_MULTI_CONFIG AND CMAKE_BUILD_TYPE STREQUAL "")
+  message(STATUS "Defaulting to Debug build as no build type was specified")
+  set(CMAKE_BUILD_TYPE "Debug")
```

Comment:
Do we want the default to be debug?

Reply:
Good question, do we?
It currently is today

Reply:
Might be something for internal design discussion. :)